plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java

```diff
@@ -2517,7 +2517,8 @@ else if (hasNaN) {
             }
         }
         // treat un-bucketed transactional table as having a single bucket on no columns
-        else if (hiveTableHandle.isInAcidTransaction()) {
+        // Note: we cannot use hiveTableHandle.isInAcidTransaction() here as transaction is not yet set in HiveTableHandle when getInsertLayout is called
```

Comment:
I'm not sure I understand this?
I can see that we do set the `transaction` in the `HiveInsertTableHandle` in `beginInsert`.

Is it because the engine calls `getInsertLayout` during planning (i.e. before execution when `beginInsert` would be called?)?

Reply:
The only other existing use I can see of the method is in https://github.com/trinodb/trino/blob/fcd6b8e4ea3f066c5cca9a3a1539f2c992b4c22c/plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveUpdatablePageSource.java#L102 which seems to be okay.

Reply:
>Is it because the engine calls getInsertLayout during planning (i.e. before execution when beginInsert would be called?)?

Yeah - exactly. We are operating on `TableHandle` which is obtained by `HiveMetadata.getTableHandle` here. And the `transaction` field is set to `NO_ACID_TRANSACTION` there.